browser/brave_ads/ads_tab_helper.h

```diff
@@ -91,6 +100,8 @@ class AdsTabHelper : public content::WebContentsObserver,
   std::vector<GURL> redirect_chain_;
   bool should_process_ = false;
 
+  SearchResultAdHandler search_result_ad_handler_;
```

Comment:
It's generally better to use unique_ptr for stuff like this so we can use forward decls in the header and have fewer external dependencies